File: C++/Easy/RuntimeError/decode-the-message_0.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        
        string alphabet = "abcdefghijklmnopqrstuvwxyz";
        string result = "";
        for(int i = 0; i < message.size(); i++){
            if(message[i] == ' '){
                result += ' ';
            } else {
                for(int j = 0; j < key.size(); j++){
                    if(key[j] == message[i]){
                        result += alphabet[j];
                        break;
                    }
                }
            }
        }
        return result;
    }
};

File: Java/Easy/WrongAnswer/decode-the-message_0.java
class Solution {
    public String decodeMessage(String key, String message) {
        
        char[] keyMap = new char[26];
        for (int i = 0; i < 26; i++) {
            keyMap[i] = ' ';
        }
        for (int i = 0; i < key.length(); i++) {
            char c = key.charAt(i);
            if (c != ' ') {
                keyMap[c - 'a'] = (char) ('a' + i);
            }
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < message.length(); i++) {
            char c = message.charAt(i);
            if (c != ' ') {
                sb.append(keyMap[c - 'a']);
            } else {
                sb.append(' ');
            }
        }
        return sb.toString();
    }
}